tests/Benchmarks/SystemTextJson/JsonReaderPerf.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BenchmarkDotNet.Attributes;
+using System;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using static Benchmarks.SystemTextJson.Helper;
+
+namespace Benchmarks.SystemTextJson
+{
+    [MemoryDiagnoser]
+    public class JsonReaderPerf
+    {
+        private byte[] data;
+        private SymbolTable symbolTable;
+        private MemoryStream mem;
+        private StreamReader reader;
+
+#pragma warning disable CS0649 // Field '' is never assigned to, and will always have its default value
+        [Params(EncoderTarget.InvariantUtf8, EncoderTarget.InvariantUtf16)]
+        public EncoderTarget Target;
+
+        // Using the string name listed in the resource file instead of the json string directly
+        // so that the benchmark output is cleaner
+        [ParamsSource(nameof(ValuesForJsonStringName))]
+        public string JsonStringName;
+#pragma warning restore CS0649
+
+        public static IEnumerable<string> ValuesForJsonStringName() => new[] { nameof(JsonStrings.HeavyNestedJson), nameof(JsonStrings.HelloWorld) };
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            symbolTable = GetTargetEncoder(Target);
+            data = EncodeTestData(Target, JsonStrings.ResourceManager.GetString(JsonStringName));
+
+            mem = new MemoryStream(data);
```

Comment:
Is it ok that this is in global setup? If we ever add new tests that use the stream, they won't be able to use it. Is there some way to initialize the stream/reader before every test?

cc: @adamsitnik